core/src/com/mygdx/game/models/HexBoardClass.java

```diff
@@ -0,0 +1,113 @@
+package com.mygdx.game.models;
+
+import java.util.*;
+
+/*
+ * The hex board of Type T that we will use as the map of the game. T will be a terrain
+ */
+public class HexBoardClass<T> implements HexBoard {
+    private Object[][] board;
+    int xLength;
+    int yLength;
+
+    public HexBoardClass(Object o, int xLength, int yLength) {
+        board = new Object[xLength][yLength];
+        this.xLength = xLength;
+        this.yLength = yLength;
+
+        for (int i = 0; i < xLength; i++) {
+            for (int j = 0; j < yLength; j++) {
+                board[i][j] = "" + i + j;
+            }
+        }
+    }
+
+    /*
+     * A function that returns the object at position x,y
+     */
+    @Override
+    public Object getHex(Position p) {
```

Comment:
Change Object to T

Reply:
You should return null if position isn't within the height and width of the board. Right now it would just throw a index out of bounds exception if someone gave a height greater than the array size.